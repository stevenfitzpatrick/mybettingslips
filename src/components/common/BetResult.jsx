import styled from 'styled-components';

import { resultToThemeMap } from '../../store/bet.constants';

const BetResult = styled.span`
  color: ${({ result, theme }) =>
    theme[resultToThemeMap[result]]
      ? theme[resultToThemeMap[result]].bg
      : '#ffffff'};
`;

export default BetResult;
