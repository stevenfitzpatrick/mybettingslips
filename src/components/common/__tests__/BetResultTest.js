import React from 'react';
import { renderWithTheme } from '@sfitzpatrick/enzyme-context-helpers';
import 'jest-styled-components';

import BetResult from '../BetResult';

describe('BetResult', () => {
  test('should load with open result props', () => {
    const tree = renderWithTheme(
      <BetResult result="Open">€100</BetResult>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should load with win result props', () => {
    const tree = renderWithTheme(
      <BetResult result="Win">+ €100</BetResult>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should load with loss result props', () => {
    const tree = renderWithTheme(
      <BetResult result="Loss">- €100</BetResult>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
