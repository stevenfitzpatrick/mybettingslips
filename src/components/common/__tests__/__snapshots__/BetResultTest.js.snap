// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetResult should load with loss result props 1`] = `
.c0 {
  color: hsl(0,57%,63%);
}

<span
  className="c0"
  result="Loss"
>
  - €100
</span>
`;

exports[`BetResult should load with open result props 1`] = `
.c0 {
  color: #ffffff;
}

<span
  className="c0"
  result="Open"
>
  €100
</span>
`;

exports[`BetResult should load with win result props 1`] = `
.c0 {
  color: hsl(152,62%,53%);
}

<span
  className="c0"
  result="Win"
>
  + €100
</span>
`;
